libsql/src/sync.rs

```diff
@@ -115,6 +120,30 @@ impl SyncContext {
 
         let durable_frame_num = self.push_with_retry(uri, frame, self.max_retries).await?;
 
+        if durable_frame_num > frame_no {
+            tracing::error!(
+                "server returned durable_frame_num larger than what we sent: sent={}, got={}",
+                frame_no,
+                durable_frame_num
+            );
+
+            // TODO: do we want to return an error here?
```

Comment:
Yes, let's error out saying that we're behind the server. This will be the place where we wire our _pull logic_ likely when the server supports it

Reply:
done